src/App.tsx

```diff
@@ -1,56 +1,80 @@
-import React from 'react';
+import React, {
+  useState,
+  useCallback,
+  useMemo,
+} from 'react';
 import './App.scss';
+import { debounce } from 'lodash';
 import { peopleFromServer } from './data/people';
+import { DropdownMenu } from './components/DropdownMenu/DropdownMenu';
+import { Person } from './types/Person';
+
+const LIST_UPDATE_DELAY = 1000;
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [isSelected, setIsSelected] = useState(false);
+  const [isFocused, setIsFocused] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, LIST_UPDATE_DELAY), [],
+  );
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(
+      person => person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery, peopleFromServer]);
+
+  function handleQueryChange(event: React.ChangeEvent<HTMLInputElement>) {
```

Comment:
it's not implemented:
![image](https://github.com/mate-academy/react_autocomplete/assets/86923089/ba1f9703-fa65-462e-9c52-25b4c9ac379a)

hint: you need a separate state value for this and change it with delay
feel free to ask in the chat if you need help with it